config/stack.go

```diff
@@ -279,12 +279,12 @@ func validateWatchPaths(rootdir string, stackpath string, paths []string) (proje
 }
 
 // StacksFromTrees converts a List[*Tree] into a List[*Stack].
-func StacksFromTrees(root string, trees List[*Tree]) (List[*SortableStack], error) {
+func StacksFromTrees(trees List[*Tree]) (List[*SortableStack], error) {
 	var stacks List[*SortableStack]
 	for _, tree := range trees {
-		s, err := NewStackFromHCL(root, tree.Node)
+		s, err := tree.Stack()
```

Comment:
@terramate-io/cli-tooling 

Note the `NewStackFromHCL()` in the line above.
The problem is that while building the DAG, stacks can be reloaded from disk and then they loose the implicit "before" clauses added for the filesystem ordering.